Introduction of 'sort by taken date'

AndyKilmory opened the pull request:
## What does this change?

The basic sort b functionality has been extended to allow the user to select to sort matching images by 'Taken Date' - either new to old or old to new.

The sort by control has been extended to include the new options...
![Screenshot 2025-04-22 at 14 40 00](https://github.com/user-attachments/assets/67d72ac9-1b8c-47f6-9443-771a2d6678e2)

Selecting  either taken date option will result in changes to the calls to the api and also the main images url...
![Screenshot 2025-04-22 at 14 41 16](https://github.com/user-attachments/assets/3316566a-9000-4b3d-9745-b6633c9ef46e)

Some images do not have a taken date and those images matching the search criteria but without a taken date will be returned at the end of the list of matches. A simple transient notification banner has been included to indicate this...
![Screenshot 2025-04-22 at 14 42 53](https://github.com/user-attachments/assets/f88ae027-7d38-40c4-b9e6-186a1e44c0e9)

In addition the summary grid display of image thumbnails has been modified so that when a 'take date' ordering has been selected the displayed date switches from 'Uploaded' to 'Taken'...
![Screenshot 2025-04-22 at 14 45 53](https://github.com/user-attachments/assets/6f165398-2d01-4d47-b8e6-7146029c2ed7)

The remainder of the system functionality should continue unaltered. Copying of URLs containing the new order by parameters should be honoured. Newly ingested images should slot into the image list in accordance with their order by values etc.

Note: this is the first PR to extend sort by to include taken date a further PR is expected to imporve the functiobality further to include better management of images without taken dates and also re-factoring of the UI to make sort by management more intuitive.

## How should a reviewer test this change?

Ensure that the sort by control behaves as expected when the new options are selected and ensure that all other functionality is unaffected by the change and that sort order is retained in the same way for taken date as for upload date

## Who should look at this?
<!-- Reach the team with @guardian/digital-cms -->

## Tested? Documented?
- [x] locally by committer
- [ ] locally by Guardian reviewer
- [ ] on the Guardian's TEST environment
- [ ] relevant documentation added or amended (if needed)

paperboyo commented:
This is amazing and long-sought for! Thank you.

Testing functionality, everything appears to be working correctly (sort order is reset on Home, correctly changes to Added to collection on Collection, survives adding, removing chips). 

These are some thoughts for consideration:
1. When I click on a collection, order correctly changes from Taken to Added to…. but when I remove collection chip then, it should go back to previous order I think (Taken if that was it)?
2. What is the secondary sort order (for images that have exactly the same, usually partial, Date Taken? I think it should be Upload Time, but it appears to be not that and I can’t tell what it is…
    ![taken](https://github.com/user-attachments/assets/5d55ef2e-f83f-437e-a700-c9dc32b597b9)
3. This is something completely unrelated and only to ponder from UX perspective, not for this PR. Some active searches are clear to users: text in searchbox, chips, checkboxes. But for some others – it’s not always clear to users that there is an active search different from default (esp. if it came with a link they clicked). This applies to eg. date filter and sort order. See below for a totally amateur way of trying to make that clearer (would apply only for when options differ from default, could be a dot or something else entirely). The idea is for users to be able to immediately tell they have some options active:
    <img width="843" alt="image" src="https://github.com/user-attachments/assets/dc86d62d-877e-4462-baad-5dd0f04ae1f3" />

Let us know what you think (only pts. 1–2 are any-relevant to this PR).

AndyKilmory commented:
Thanks for the feedback Mateusz - I'll look to implement points 1 and 2 and talk with the team regarding point 3.

AndyKilmory commented:
Have addressed points 1 and 2 with recent revisions. One thing I would say is that the UI doesn't show the full value held for 'taken date' in some cases - it shows to minute only - whilst the sort works on the fully resolved value - so a number of pictures can appear to have the same Taken Date in the UI but still doesn't order according to Upload Date. The sort on 'taken date' is resolving on data (seconds etc) not shown in the UI. Where the Taken Dates are identical then Upload Date will be the secondary sort. Hope that's clear. Thanks

paperboyo commented:
Thanks for all the fixes, I tested and confirm secondary sort order (uploadTime) works correctly, coming back from teh Collection sort works well and thanks too for more granularity in tooltips.

This is now ready for a formal review.

paperboyo commented:
Also, this fixes https://github.com/guardian/grid/issues/1925.
